android/src/main/java/com/tailscale/ipn/QuickToggleService.java

```diff
@@ -114,13 +114,14 @@ public void onClick() {
     }
 
     private void onTileClick() {
-        boolean act;
+        boolean needsToStop;
         synchronized (lock) {
-            act = getAbleToStartVPN(this) && isRunning;
+            needsToStop = getAbleToStartVPN(this) && isRunning;
+        }
+        if (needsToStop) {
+            App.getApplication().stopVPN();
```

Comment:
We can reuse the applications stop and start methods instead of building intents by hand.